### PR TITLE
refactor: reduce usage of chainstate globals in Dash-specific logic

### PR DESCRIPTION
**Pull request by kwvg**

### CODE REVIEW

File: src/evo/deterministicmns.h [hunk 1]
```diff
@@ -582,14 +583,15 @@ class CDeterministicMNManager
...
 
     std::unordered_map<uint256, CDeterministicMNList, StaticSaltedHasher> mnListsCache GUARDED_BY(cs);
     std::unordered_map<uint256, CDeterministicMNListDiff, StaticSaltedHasher> mnListDiffsCache GUARDED_BY(cs);
     const CBlockIndex* tipIndex GUARDED_BY(cs) {nullptr};
 
 public:
-    explicit CDeterministicMNManager(CEvoDB& evoDb, CConnman& _connman) :
-        m_evoDb(evoDb), connman(_connman) {}
+    explicit CDeterministicMNManager(CChainState& chainstate, CConnman& _connman, CEvoDB& evoDb) :
+        m_chainstate(chainstate), connman(_connman), m_evoDb(evoDb) {}
```
[the remark this answers is not in the document]
Comment (knst) [reply src/evo/deterministicmns.h:594]: btw, created PR to let it catch by CI for further PRs. https://github.com/dashpay/dash/pull/5540

File: src/init.cpp [hunk 2]
```diff
@@ -1937,12 +1937,13 @@ bool AppInitMain(const CoreContext& context, NodeContext& node, interfaces::Bloc
 
                 // Same logic as above with pblocktree
                 deterministicMNManager.reset();
-                deterministicMNManager.reset(new CDeterministicMNManager(*node.evodb, *node.connman));
+                deterministicMNManager.reset(new CDeterministicMNManager(chainman.ActiveChainstate(), *node.connman, *node.evodb));
+                creditPoolManager.reset();
```
[the remark this answers is not in the document]
Comment (kwvg) [reply src/init.cpp:1941]: From the comment in f8fbb32f8db0710658e56f61157db79045560812:

> See https://github.com/bitcoin/bitcoin/pull/12401 and surrounding discussion

File: src/test/util/setup_common.cpp [hunk 3]
```diff
@@ -95,6 +95,26 @@ std::ostream& operator<<(std::ostream& os, const uint256& num)
     return os;
 }
 
+namespace dash {
```
Comment (PastaPastaPasta) [inline src/test/util/setup_common.cpp:98]: can we please come up with a better name here?

File: src/evo/deterministicmns.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (kwvg) [reply src/evo/deterministicmns.h:594]: Order of members have been changed to match the order of initialization

File: src/test/validation_chainstatemanager_tests.cpp [hunk 4]
```diff
@@ -4,14 +4,17 @@
 //
 #include <chainparams.h>
 #include <consensus/validation.h>
+#include <evo/deterministicmns.h>
 #include <index/txindex.h>
 #include <llmq/blockprocessor.h>
 #include <llmq/chainlocks.h>
+#include <llmq/context.h>
```
[the remark this answers is not in the document]
Comment (kwvg) [reply src/test/validation_chainstatemanager_tests.cpp]: Remnants of pre-consolidated Dash components init, should've been removed with e8ca60f844051957eec6f2ff261498b90cd04d7f. Resolved in latest puish.

File: src/test/util/setup_common.cpp (hunk 3 above)
Comment (kwvg) [reply src/test/util/setup_common.cpp:98]: Initialization logic isn't limited to LLMQ, masternodes more or less are their own subsystem, even if they interact heavily with everything else, the only thing encompassing them all is that they're Dash-specific.

Comment (knst) [reply src/test/util/setup_common.cpp:98]: so far as it used only in src/test/util/setup_common.cpp and `validation_chainstatemanager_tests.cpp` (only RegTest) and doesn't used inside init.cpp for normal app initialization, let's give it more specific name? Such as `DashRegnetInit` or `DashTestSetup` or something similar.


Btw, I don't like direct call of `dash::DestroyComponents`

Comment (kwvg) [reply src/test/util/setup_common.cpp:98]: Neither do I but the chainstates created affect `m_node`, affecting the scope of the entire function and Dash's subsystems are tied to a chainstate, so if they're torn down and rebuilt, we cannot reuse the objects initialized on the older chainstate.

Naming them `DashTestSetup` and `DashTestSetupClose` works for me. Thoughts @PastaPastaPasta?

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Pull request merged**